pkg/kubelet/cm/deviceplugin/manager.go

```diff
@@ -73,8 +73,11 @@ type ManagerImpl struct {
 	// e.g. a new device is advertised, two old devices are deleted and a running device fails.
 	callback monitorCallback
 
-	// allDevices contains all of registered resourceNames and their exported device IDs.
-	allDevices map[string]sets.String
+	// healthyDevices contains all of the registered healthy resourceNames and their exported device IDs.
+	healthyDevices map[string]sets.String
+
+	// unhealthyDevices contains all of the unhealthy devices and their exported device IDs.
+	unhealthyDevices map[string]sets.String
```

Comment:
I'm just wondering is there any benefits we keep `unhealthyDevices` in cache here. Maybe we can make more discussions about this :)

Reply:
Kubelet needs to update node status taking into account unhealthy devices also(in the capacity). If we dont store unhealthy devices here in device manager, i am not sure how kubelet will sync this info. Would love to hear any suggestions?
btw, this PR is going to changed heavily after once #56398 is merged. I am waiting for that to get merged.